bwpy/signal.py

```diff
@@ -0,0 +1,133 @@
+from . import mea_viewer
+import re
+import abc
+import functools
+import numpy as np
+import numpy.lib.stride_tricks as np_tricks
+
+
+__all__ = []
+
+
+def _transformer_factory(cls):
+    @functools.wraps(cls)
+    def transformer_factory(slice, *args, **kwargs):
+        return slice._transform(cls(*args, **kwargs))
+
+    return transformer_factory
+
+
+class Transformer(abc.ABC):
+    def __init_subclass__(cls, operator=None, **kwargs) -> None:
+        super().__init_subclass__(**kwargs)
+        name = operator or re.sub("([a-z0-9])([A-Z])", r"\1_\2", cls.__name__).lower()
+        globals()[name] = _transformer_factory(cls)
+        __all__.append(name)
+
+    def _apply_window(self, data, bin_size):
+        rows = data.shape[1]
+        cols = data.shape[2]
+        window_shape = (bin_size, rows, cols)
+        # sliding window returns more complex shape like (num_windows, 1, 1, rows, cols, bin_size)
+        # with the reshape we get rid of the unnecessary complexity (1, 1)
+        return np_tricks.sliding_window_view(data, window_shape).reshape(
+            -1, bin_size, rows, cols
+        )
+
+    @abc.abstractmethod
+    def __call__(self, data, file):
+        pass
+
+
+class Variation(Transformer):
+    def __init__(self, bin_size):
+        self.bin_size = bin_size
+
+    def __call__(self, data, slice, file):
+        print("calling ", self.__class__.__name__)
+        if data.ndim < 3:
+            return data
+        else:
+            windows = self._apply_window(data, self.bin_size)
+            return np.max(np.abs(windows), axis=1) - np.min(np.abs(windows), axis=1)
+
+
+class Amplitude(Transformer):
+    def __init__(self, bin_size):
+        self.bin_size = bin_size
+
+    def __call__(self, data, slice, file):
+        print("calling ", self.__class__.__name__)
+        if data.ndim < 3:
+            return data
+        else:
+            windows = self._apply_window(data, self.bin_size)
+            return np.max(np.abs(windows), axis=1)
+
+
+class Energy(Transformer):
+    def __init__(self, bin_size):
+        self.bin_size = bin_size
+
+    def __call__(self, data, slice, file):
+        print("calling ", self.__class__.__name__)
+        if data.ndim < 3:
+            return data
+        else:
+            windows = self._apply_window(data, self.bin_size)
+            return np.sum(np.square(windows), axis=1)
+
+
+class Raw(Transformer):
+    def __call__(self, data, slice, file):
+        print("calling ", self.__class__.__name__)
+        return np.moveaxis(data, 2, 0)
+
+
+class NoMethod(Transformer):
+    def __call__(self, data, slice, file):
+        print("calling ", self.__class__.__name__)
+        return data
```

Comment:
```suggestion
class Noop(Transformer):
    """Noop that doesn't transform the data at all."""
    def __call__(self, data, slice, file):
        return data
```

Noop (no-operation) is a more conventional name